### PR TITLE
refactor: database decorators

### PR DESCRIPTION
**Pull request by geolffreym**
- added to `files` package `read`, and `exists` functions
- added to `cache` package database connection and decorators
- splitting `storage` from  `ipfs` package (SRP)

Note: please omit anything not related to above described

### CODE REVIEW

**Comment by geolffreym** [comment]
Screenshot from 2022-08-23 11-22-08

File: src/core/cache/manager.py [hunk 1]
```diff
@@ -1,109 +1,105 @@
...
-    Just remove old data and replace it with new data in temp db
-    :param data:
-    """
-    try:
-        raw_db.movies.delete_many({})  # Clean all
-        raw_db.movies.insert_many(data)
-    except BulkWriteError:
-        log.error("Failed to rewrite data")
+
+# TODO add here generic methods to interact with cache database
```
Comment (aphelionz) [inline src/core/cache/manager.py:2]: Should we just delete this file?

**Review by aphelionz — APPROVED** [review]
Just one question

File: src/core/cache/manager.py (hunk 1 above)
Comment (geolffreym) [reply src/core/cache/manager.py:2]: Definitely. i am wondering if there we can add some "core level" query or generic functions? @aphelionz

**Pull request merged**